kie-maven-plugin/src/main/java/org/kie/maven/plugin/ValidateDMNMojo.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.maven.plugin;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.drools.compiler.kproject.models.KieModuleModelImpl;
+import org.kie.api.builder.Message.Level;
+import org.kie.api.builder.model.KieModuleModel;
+import org.kie.dmn.api.core.DMNMessage;
+import org.kie.dmn.core.assembler.DMNAssemblerService;
+import org.kie.dmn.core.compiler.DMNProfile;
+import org.kie.dmn.feel.util.ClassLoaderUtil;
+import org.kie.dmn.model.api.DMNModelInstrumentedBase;
+import org.kie.dmn.model.api.Definitions;
+import org.kie.dmn.validation.DMNValidator;
+import org.kie.dmn.validation.DMNValidator.Validation;
+import org.kie.dmn.validation.DMNValidatorFactory;
+import org.kie.internal.utils.ChainedProperties;
+
+@Mojo(name = "validateDMN",
+      requiresDependencyResolution = ResolutionScope.NONE,
+      requiresProject = true,
+      defaultPhase = LifecyclePhase.PROCESS_RESOURCES)
+public class ValidateDMNMojo extends AbstractKieMojo {
+
+    @Parameter(required = true, defaultValue = "${project.build.resources}")
+    private List<Resource> resources;
+
+    @Parameter
+    private Map<String, String> properties;
+
+    @Parameter(required = true, defaultValue = "${project}")
+    private MavenProject project;
+
+    @Parameter(property = "validateDMN", defaultValue = "VALIDATE_SCHEMA,VALIDATE_MODEL")
+    private String validateDMN;
+    private List<Validation> actualFlags = new ArrayList<>();
+
+    private DMNValidator validator;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
```

Comment:
That's a big function. Could it be broken in small methods to be better tested? My other concern is because I can't see a test for this new class. In case I'm wrong, could you please write a quick explanation? Thanks.

Reply:
> please note tests are provided by takari framework here: https://github.com/kiegroup/droolsjbpm-integration/tree/master/kie-plugins-testing/src/test/projects/kjar-7-with-dmn

If I'm not wrong the test as per all the other Mojos (like DRL etc) are provided by means of Takari framework and the downstream build.

Reply:
Thanks @tarilabs

Reply:
@tarilabs 
Do I understand correctly that we are adding validator to validate DMN during plugin execution? What about to add test for invalid DMN to https://github.com/kiegroup/droolsjbpm-integration/tree/master/kie-plugins-testing/src/test/projects/kjar-7-with-dmn ? It will test that validator throws error when DMN is invalid.

Reply:
I need to create a separate test for that, because it would be a Maven Failed build if not passing the validation for an Error, so it would change semantic for existing test. Do you concur it would be wrong to do it on an existing test which semantic is expected to be Maven Success build?

Reply:
Also @jiripetrlik if you want me to create a test for a Maven Failure build, can you kindly point me to an example of a test showing a Maven Failure, please? So far all the tests have Maven Success semantic unless I missed something.